apps/ejabberd/src/mod_http_notification.erl

```diff
@@ -15,38 +15,17 @@
 -include("ejabberd.hrl").
 -include("jlib.hrl").
 
--define(DEFAULT_HTTP_POOL_SIZE, 100).
--define(DEFAULT_HTTP_HOST, "http://localhost").
--define(DEFAULT_PREFIX_PATH, <<"/">>).
--define(DEFAULT_HTTP_WORKER_TIMEOUT, 5000).
--define(DEFAULT_HTTP_POOL_TIMEOUT, 200).
+-define(DEFAULT_POOL_NAME, http_pool).
+-define(DEFAULT_PATH, <<"">>).
 
 start(Host, _Opts) ->
-    HttpHost = gen_mod:get_module_opt(Host, ?MODULE, host, ?DEFAULT_HTTP_HOST),
-    PoolSize = gen_mod:get_module_opt(Host, ?MODULE, pool_size, ?DEFAULT_HTTP_POOL_SIZE),
-    Timeout = gen_mod:get_module_opt(Host, ?MODULE, worker_timeout, ?DEFAULT_HTTP_WORKER_TIMEOUT),
-    PathPrefix = gen_mod:get_module_opt(Host, ?MODULE, prefix_path, ?DEFAULT_PREFIX_PATH),
-    PoolName = pool_name(Host),
-    PoolOpts = [{name, {local, PoolName}},
-                {size, PoolSize},
-                {max_overflow, 5},
-                {worker_module, mod_http_notification_requestor}],
-    WorkerOpts = [{http_host, HttpHost},
-                  {timeout, Timeout},
-                  {path_prefix, PathPrefix}],
-    {ok, _} = supervisor:start_child(ejabberd_sup,
-        {PoolName,
-            {poolboy, start_link,
-                [PoolOpts, WorkerOpts]},
-            transient, 2000, supervisor, [mod_http_notification_requestor]}),
+    Path = gen_mod:get_module_opt(Host, ?MODULE, path, ?DEFAULT_PATH),
+    PoolName = gen_mod:get_module_opt(Host, ?MODULE, pool_name, ?DEFAULT_POOL_NAME),
+    mod_http_notification_requestor:start(Host, PoolName, Path),
```

Comment:
It seems that this `requestor` may be not needed at all - is there any reason to keep that extra `gen_server`?